cmd/faFindFast/efficient.go

```diff
@@ -120,3 +124,25 @@ func speedyWindowDifference(windowSize int, reference []dna.Base, query []dna.Ba
 		}
 	}
 }
+
+// binomialDistCacheLog10 generates a map of form map[int]float64 for a binomial distribution specified by parameters n and k.
+// map[k] returns a float64 representing -log10(BinomialDist(n, k, p, true)), or the -log10pValue.
+func binomialDistCacheLog10(n int, p float64) map[int]float64 {
+	if p < 0 || p > 1 {
+		log.Fatalf("Error: p must be a value between 0 and 1. Found: %v.\n", p)
+	}
+	var answer = make(map[int]float64)
+	answer[n] = numbers.BinomialDistLog(n, n, p)
+
+	for k := n - 1; k >= 0; k-- {
+		answer[k] = logspace.Add(numbers.BinomialDistLog(n, k, p), answer[k+1])
+	}
+
+	// now convert to -log10 space
+	for k := 0; k <= n; k++ {
+		answer[k] = -1 * logspace.ToBase10(answer[k])
+	}
+	answer[0] = 0 //hardcoded to avoid numerical noise
```

Comment:
How much is this noise?  Maybe there should be a check here that if answer[0] is too far from 1 or 0 (depending on when the check happens) that we die.  Maybe we check to make sure it is between 0.99 and 1.01?

Reply:
I tested without this line and the answer I got back was 3.616228e-17. In other words, very minor noise